xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1225,7 +1225,8 @@ namespace VIDEO
         int offset = 0;
 
         // we want "long circuit" OR below so that both offsets are evaluated
-        while (((regexp2pos = reg2.RegFind(remainder.c_str() + offset)) > -1) | ((regexppos = reg.RegFind(remainder.c_str() + offset)) > -1))
+        while (static_cast<int>((regexp2pos = reg2.RegFind(remainder.c_str() + offset)) > -1) |
+               static_cast<int>((regexppos = reg.RegFind(remainder.c_str() + offset)) > -1))
```

Comment:
it may silence the warning and all but what do you think about this?
```c
regexppos = reg.RegFind(remainder.c_str() + offset);
regexp2pos = reg2.RegFind(remainder.c_str() + offset);

while (regexppos > -1 || regexp2pos > -1)
{
  // DoStuff

  regexppos = reg.RegFind(remainder.c_str() + offset);
  regexp2pos = reg2.RegFind(remainder.c_str() + offset);
}
```
it is your choice, thanks for fixing

Reply:
I prefer to keep the original code, it's an elegant solution and we don't touch the logic.

Reply:
cool